src/transformers/generation_dist_process.py

```diff
@@ -0,0 +1,312 @@
+# limitations under the License.
+
+from abc import ABC
+from typing import Iterable, List
+
+import numpy as np
+import torch
+from torch.nn import functional as F
+
+from .file_utils import add_start_docstrings_to_callable
+
+
+DIST_PROCESSOR_INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`):
+            Indices of input sequence tokens in the vocabulary.
+
+            Indices can be obtained using :class:`~transformers.BertTokenizer`. See
+            :meth:`transformers.PreTrainedTokenizer.encode` and :meth:`transformers.PreTrainedTokenizer.__call__` for
+            details.
+
+            `What are input IDs? <../glossary.html#input-ids>`__
+        scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, config.vocab_size)`):
+            Prediction scores of a language modeling head. These can be scores for each vocabulary token before SoftMax
+            or scores for each vocabulary token after SoftMax.
+
+    Returns:
+        scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, config.vocab_size)`): The processed prediction
+        scores.
+"""
+
+
+class DistProcessorList(list):
+    """
+    This class can be used to create a list of :class:`~transformers.DistProcessor` to subsequently process a
+    :obj:`scores` input tensor. This class inherits from list and adds a specific `__call__` method to apply each
+    :class:`~transformers.DistProcessor` to the inputs.
+    """
+
+    @add_start_docstrings_to_callable(DIST_PROCESSOR_INPUTS_DOCSTRING)
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
+        for processor in self:
+            scores = processor(input_ids, scores)
+        return scores
+
+
+class DistProcessor(ABC):
+    """Abstract base class for all samplers which are probability distribution warps performed during generation."""
+
+    @add_start_docstrings_to_callable(DIST_PROCESSOR_INPUTS_DOCSTRING)
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
+        """Torch method for processing a distribution."""
+        raise NotImplementedError(
+            f"{self.__class__} is an abstract class. Only classes inheriting this class can be called."
+        )
+
+
+class MinLengthDistProcessor(DistProcessor):
+    """DistProcessor enforcing a min-length by setting EOS probability to 0."""
+
+    def __init__(self, min_length: int, eos_token_id: int):
+        if not isinstance(min_length, int) or min_length < 0:
+            raise ValueError(f"`min_length` has to be a positive integer, but is {min_length}")
+
+        if not isinstance(eos_token_id, int) or eos_token_id < 0:
+            raise ValueError(f"`eos_token_id` has to be a positive integer, but is {eos_token_id}")
+
+        self.min_length = min_length
+        self.eos_token_id = eos_token_id
+
+    @add_start_docstrings_to_callable(DIST_PROCESSOR_INPUTS_DOCSTRING)
```

Comment:
You don't need to put the docstring here again unless you add modifications, it will be inherited from the superclass.